host_pool.js

```diff
@@ -56,26 +56,26 @@ function HostPool (hostports_str, retry_secs) {
  * timer.
  */
 HostPool.prototype.failed = function (host, port) {
-    var self = this;
-    var key = host + ':' + port;
-    var retry_msecs = self.retry_secs * 1000;
+    const self = this;
+    const key = host + ':' + port;
+    const retry_msecs = self.retry_secs * 1000;
     self.dead_hosts[key] = true;
 
-    var cb_if_still_dead = function (){
+    function cb_if_still_dead () {
         logger.logwarn("host " + key + " is still dead, will retry in " +
                         self.retry_secs + " secs");
         self.dead_hosts[key] = true;
         // console.log(1);
         setTimeout(function () {
             self.probe_dead_host(host, port, cb_if_still_dead, cb_if_alive);
         }, retry_msecs);
-    };
+    }
 
-    var cb_if_alive = function (){
+    function cb_if_alive () {
```

Comment:
Keep declarations consistent

Reply:
Not necessarily. If you don't need variable hoisting, why maintain this unusual style of declaration?

Reply:
The weird way that js starts setting functions is that if they're defined with `function x` means it's everywhere even if the block it's contained in doesn't run.

Whereas if you define them as a `var`, `let`, or `const`, you can have a function named `process` within two different blocks of code and they wouldn't interfere with each other.

It's just keeping it cleaner and nothing messes with each other or introduces any weird bugs because someone's defined `function x` in two different blocks of code.

http://adripofjavascript.com/blog/drips/variable-and-function-hoisting

Reply:
> The weird way that js starts setting functions is that if they're defined with function x means it's everywhere even if the block it's contained in doesn't run.

That's not quite how hoisting works. Function declarations aren't available **everywhere** unless you declare them in global scope. Function declarations are only hoisted to the top of their current block/scope. See the links I posted below which go into much greater details. `var` is hoisted the same way, but `let` and `const` are not. Hence, prior to this change, a function expression using var was hoisted the same as a function declaration. When you switch to const/let, you lose the hoisting entire, creating the infamous Temporal Dead Zone. Again, the links below delve into the gritty details.